run/markdown-preview/editor/main.js

```diff
@@ -0,0 +1,81 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+const express = require('express');
+const handlebars = require('handlebars');
+const { readFileSync } = require('fs');
+const renderRequest = require('./render.js');
+
+const app = express();
+app.use(express.json());
+
+let url, isAuthenticated, markdownDefault, compiledTemplate, template;
+
+const init = () => {
```

Comment:
I know we've gone a couple rounds on this, however I just wanted to flag that configuration loading is often considered it's own dedicated thing. In a larger service, I wouldn't want to include all of this in the same file:

* Create a server
* Load configuration
* Define the routes/controllers

Reply:
In light of no longer needing the `EDITOR_UPSTREAM_UNAUTHENTICATED` env var with the new auth ID token method, I agree that the `init()` function should be removed entirely. The `EDITOR_UPSTREAM_RENDER_URL` is used only in /editor/render, so I'm going to put the logic to check for and use that env var in /editor/render. (Let me know if that's not the right idea.)